File: 프로그래머스/2/389479. 서버 증설 횟수/서버 증설 횟수.py
def solution(players, m, k):
    answer = 0
    table = [0] * (24+k)
    server = 0
    for i in range(24):
        server_down = table[i]
        if server_down:
            server -= server_down
        
        need = players[i] // m
        if need > server:
            server_on = need - server
            answer += server_on
            server += server_on
            table[i+k] += server_on

    return answer

File: 프로그래머스/2/389479. 서버 증설 횟수/서버 증설 횟수.java
import java.util.*;

class Solution {
    public int solution(int[] players, int m, int k) {
        int answer = 0;
        int server = 0;
        int[] table = new int[24 + k];
        
        // System.out.println(Arrays.toString(table));
        
        for(int time = 0 ; time < 24 ; time ++){
            int serverDown = table[time];
            if(serverDown > 0){
                server -= serverDown;
            }
            
            int oper = 0;
            int need = players[time] / m;
            if (need > server){
                oper = need - server;
                server += oper;
                table[time + k] += oper;
                answer += oper;
                // System.out.println("oper:" + oper);
                
            }
            // System.out.print(players[time]);
            // System.out.print(server);
            // System.out.println(oper);
            // System.out.println("time:" + time);
            // System.out.println("need:" + need);
            // System.out.println();
        }
        return answer;
    }
}